events/wc_undead_events.txt

```diff
@@ -1598,12 +1857,34 @@ character_event = {
 		name = EVTOPTA_WCUND_375
 
 		remove_racial_traits_effect = yes
-		add_trait = creature_abomination
+		
+		# Assign creature trait depending on religion		
+		if = {
+			limit = { religion = throsic } 
+		}
+		add_trait = creature_wicker_golem
+
+		else = {
+			add_trait = creature_abomination
+		}
 
 		# Fixes immortal age
 		add_age_back_effect = yes
 
+		# Set graphical culture depending on creature type
+		if = {
+			limit = {
+				trait = creature_abomination
+			}
+		}
 		hidden_effect = { set_graphical_culture = abomination }
+		
+		if = {
+			limit = {
+				trait = creature_wicker_golem
+			}
+		}
+		hidden_effect = { set_graphical_culture = wicker_golem }
```

Comment:
Doesn't work because the graphical culture is outside the if blocks.
```suggestion
		if = {
			limit = { trait = creature_abomination }
			hidden_effect = { set_graphical_culture = abomination }
		}
		if = {
			limit = { trait = creature_wicker_golem }
			hidden_effect = { set_graphical_culture = wicker_golem }
		}
```